### PR TITLE
build-script: add missing DocC dependency

### PR DESCRIPTION
**Pull request by MaxDesiatov**
Without this dependency, passing both `--swiftdocc` and `--install-swiftdocc` to `build-script` causes it to crash with this stack trace:

```
Traceback (most recent call last):
  File "./swift/utils/build-script", line 789, in <module>
    exit_code = main()
  File "./swift/utils/build-script", line 784, in main
    return main_normal()
  File "./swift/utils/build-script", line 740, in main_normal
    invocation.execute()
  File "./swift/utils/swift_build_support/swift_build_support/build_script_invocation.py", line 670, in execute
    (self.impl_env, self.impl_args) = self.convert_to_impl_arguments()
  File "./swift/utils/swift_build_support/swift_build_support/build_script_invocation.py", line 147, in convert_to_impl_arguments
    for product_class in sum(list(self.compute_product_pipelines()[0]), []):
  File "./swift/utils/swift_build_support/swift_build_support/build_script_invocation.py", line 664, in compute_product_pipelines
    return builder.finalize(shouldInfer=self.args.infer_dependencies)
  File "./swift/utils/swift_build_support/swift_build_support/productpipeline_list_builder.py", line 198, in finalize
    result = self.infer()
  File "./swift/utils/swift_build_support/swift_build_support/productpipeline_list_builder.py", line 157, in infer
    build_graph.produce_scheduled_build(enabled_pipeline)[0]
  File "./swift/utils/swift_build_support/swift_build_support/build_graph.py", line 150, in produce_scheduled_build
    dag.set_root(entry)
  File "./swift/utils/swift_build_support/swift_build_support/build_graph.py", line 95, in set_root
    assert self.root is None
AssertionError
```

Apparently, the absence of this explicit dependency makes `build_graph.py` code think that `SwiftDocCRender` and `SwiftDocC` are independent root targets, which it can't handle. Making one a dependency of the other resolves the issue.

### CODE REVIEW

**Comment by MaxDesiatov** [comment]
@swift-ci please smoke test

**Comment by MaxDesiatov** [comment]
@swift-ci please smoke test

File: utils/swift_build_support/README.md [hunk 1]
```diff
@@ -6,5 +6,5 @@ structures used by the Swift build script.
 You may run unit tests for `swift_build_support` from the command line:
 
 ```sh
-$ python utils/swift_build_support/run_tests.py
```
Comment (etcwilde) [inline utils/swift_build_support/README.md:9]: While a reasonable fix, this change looks unrelated

**Review by etcwilde — APPROVED** [review]
LGTM

**Review by ethan-kusters — APPROVED** [review]

File: utils/swift_build_support/README.md (hunk 1 above)
Comment (MaxDesiatov) [reply utils/swift_build_support/README.md]: makes sense, I'll submit it as a separate PR

**Comment by MaxDesiatov** [comment]
@swift-ci please test Windows
